Extend Timeout for Events Verification on E2E Tests

razo7 opened the pull request:
<!-- Thanks for contributing to our project! We appreciate your time and effort.
Please fill out the information below to expedite the review and merge of your pull request.
-->

#### Why we need this PR
<!--
Outline the purpose of this PR, whether it's addressing a bug, implementing a new feature, or solving a specific problem.
-->

- Fix the Flaky E2E test regarding an error in the missing succeeded maintenance event.
- Extend the verification timeout of existing/non-existing events from 120 sec to 180 sec, since successful event might happen after 2 minutes.

#### Changes made
<!-- Outline the specific changes made in this merge request. -->

- Extending events timeout from 120 sec to 180 sec.
- Small fixes and fixes to deprecated modules

#### Which issue(s) this PR fixes
<!--
Any reference to relevant issue(s).
Please use the following format, so that the issue will be automatically closed when this PR is merged (see https://help.github.com/articles/closing-issues-using-keywords/)

`Fixes #<issue number>`

If there is not a correspondent issue yet, you might want to open a new one yourself, describing the problem you observed.
-->


#### Test plan
<!--
Please, make sure that this PR meets all the necessary quality gates before submitting for review:

- Existing Unit and E2E tests are passing
- New features or bug fixes should be covered by new Unit and/or E2E tests.

This will help us to ensure that your changes are working as expected and will not break in the future.
 
In order to save cloud resources, we invite you to submit the PR as a draft and run a single E2E test job, e.g. adding a comment to the PR with the following message in order to run E2E test on OCP 4.15 only:

> /test 4.15-openshift-e2e

In case you are unable to verify E2E test prior to submit the PR, we suggest to use the WIP (Work In Progress) title prefix (e.g. "[WIP] <Title of the PR>"), and then follow the above mentioned manual test steps. Once the E2E job passes, you can remove the WIP prefix and request a review.
-->

razo7 commented:
/test 4.17-openshift-e2e

razo7 commented:
/test 4.17-openshift-e2e

razo7 commented:
/test 4.18-openshift-e2e

slintes commented on test/e2e/node_maintenance_test.go:
```diff
@@ -35,6 +35,8 @@ const (
 	eventInterval         = time.Second * 10
 	timeout               = time.Second * 120
 	testDeployment        = "test-deployment"
+	// sometimes an event is emitted so quic that it is raced with the the current time which is
```
I doubt that quick events are the reason. There should be enough time between creating a CR, and the controller reconciling it and emitting the event. I think it's more likely that the time on the controller pod and the test pod are out of sync...?
Also: what about adding the fix to the waitForEvent() function only, instead of everywhere it's called? When we have it at one place only, we also don't need this const with that weird name ;)

razo7 replied:
>I think it's more likely that the time on the controller pod and the test pod are out of sync?

I have checked it locally with the following commands and on AWS cluster bot and there is no significant out-of-sync timing between them.

> ➜   oc exec -it node-maintenance-operator-controller-manager-5bcf79f46-wbpn4 -n openshift-workload-availability -- date -u
Thu Jan 30 15:55:08 UTC 2025
➜ oc exec -it test-deployment-856597466f-j45cs -n node-maintenance-test -- date -u                                      
Thu Jan 30 15:55:12 UTC 2025

razo7 replied:
>Also: what about adding the fix to the waitForEvent() function only, instead of everywhere it's called? When we have it at one place only, we also don't need this const with that weird name ;)

I tried it and it also worked without having this subtle delay..

razo7 commented:
/test 4.17-openshift-e2e
/test 4.18-openshift-e2e

slintes commented on test/e2e/node_maintenance_test.go:
```diff
@@ -472,30 +474,32 @@ func waitForEvent(ctx context.Context, eventReason, eventIdentifier string) erro
 		for _, event := range events.Items {
 			if strings.Contains(event.Name, eventIdentifier) && event.Reason == eventReason {
-				return true, nil
+				if event.LastTimestamp.Time.After(beginTime) {
```
oh, somehow I missed that startTime / beginTime (why 2 different names) is completely new... so... I'm wondering: we **add** new check to the conditions, and it will work better than without that check? 🤔

slintes replied:
Looking at the events of a failed job, there really is no 2nd succeeded event 🤷🏼‍♂️ 
Look for "reportingComponent": "NodeMaintenance"
Heads up, big file 😉 

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/medik8s_node-maintenance-operator/143/pull-ci-medik8s-node-maintenance-operator-main-4.17-openshift-e2e/1884533178106908672/artifacts/openshift-e2e/gather-extra/artifacts/events.json

Also, see NHC logs here, no logs for sending the event for the 2nd maintenance, looks like pod eviction doesn't finish in time: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/medik8s_node-maintenance-operator/143/pull-ci-medik8s-node-maintenance-operator-main-4.17-openshift-e2e/1884533178106908672/artifacts/openshift-e2e/gather-extra/artifacts/pods/nmo-install_node-maintenance-operator-controller-manager-6cc47455d7-szf4l_manager.log

So probably something I usually try to avoid might help here: increase `timeout`
(and you can remove that starttime check...)

slintes replied:
> Yes, but on the last succeeded test there are two succeded events

I was referring to the failed tests

> so in practice, it did help somehow

Why do you think so? IMHO that's impossible

razo7 commented:
/test 4.17-openshift-e2e
/test 4.18-openshift-e2e

razo7 commented:
/test 4.17-openshift-e2e
/test 4.18-openshift-e2e

Now trying without the event's time check, and with only increasing the timeout

razo7 committed:
Small fixes for depreciation and naming

k8s.io/utils/pointer is depreciated and replaced with k8s.io/utils/ptr

razo7 committed:
Increase timeout for verifying events

Increase timeout for verifying events. Waiting for the success event, and evicting all the pods, may take more time than 120 seconds

razo7 commented:
/test 4.17-openshift-e2e
/test 4.18-openshift-e2e

ops, I forgot to add the increase in timeout commit. Now it is included

slintes commented:
/lgtm
/test all

razo7 commented:
/cherry-pick release-0.18
